src/mono/netcore/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -10,26 +10,49 @@
 
 namespace System.Threading
 {
-    public sealed class RegisteredWaitHandle : MarshalByRefObject
+    public sealed partial class Thread
     {
-        internal RegisteredWaitHandle(WaitHandle waitHandle, _ThreadPoolWaitOrTimerCallback callbackHelper,
-            int millisecondsTimeout, bool repeating)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void ResetThreadPoolThread()
         {
+            Debug.Assert(this == CurrentThread);
+            Debug.Assert(IsThreadPoolThread);
+
+            if (_mayNeedResetForThreadPool)
+            {
+                ResetThreadPoolThreadSlow();
+            }
         }
+    }
 
+    public sealed partial class RegisteredWaitHandle : MarshalByRefObject
+    {
         public bool Unregister(WaitHandle? waitObject)
         {
             throw new PlatformNotSupportedException();
         }
     }
 
+    internal sealed partial class CompleteWaitThreadPoolWorkItem : IThreadPoolWorkItem
+    {
+        void IThreadPoolWorkItem.Execute()
+        {
+            Debug.Fail("Registered wait handles are currently not supported");
+        }
+    }
+
     public static partial class ThreadPool
     {
+        internal const bool SupportsTimeSensitiveWorkItems = true;
         internal const bool EnableWorkerTracking = false;
 
         private static bool _callbackQueued;
 
-        internal static void InitializeForThreadPoolThread() { }
+        internal static bool CanSetMinIOCompletionThreads(int ioCompletionThreads) => true;
+        internal static void SetMinIOCompletionThreads(int ioCompletionThreads) { }
+
+        internal static bool CanSetMaxIOCompletionThreads(int ioCompletionThreads) => true;
```

Comment:
Seems this should be `false`

Reply:
Hmm actually I don't think these methods are needed in this version. It's only called from the portable thread pool implementation to check validity of the value passed in by the user. I'll remove.